src/Network/Minio/Data.hs

```diff
@@ -182,6 +182,47 @@ type Region = Text
 -- APIs.
 type ETag = Text
 
+-- |
+-- Data type represents various options specified for PutObject call
+data PutObjectOptions = PutObjectOptions {
+    pooContentType :: Maybe Text
+  , pooContentEncoding :: Maybe Text
+  , pooContentDisposition :: Maybe Text
+  , pooCacheControl :: Maybe Text
+  , pooUserMetadata :: [(Text, Text)]
+  , pooNumThreads :: Maybe Word
+    } deriving (Show, Eq)
+
+-- We expose the data type PutObjectOptions (but not the
+-- constructor of the same name), and we expose all the members.
```

Comment:
The above comment lines can be removed, if you add some more helpful haddock documentation for `PutObjectOptions` - for e.g. see https://www.stackage.org/haddock/lts-9.18/http-client-0.5.7.1/Network-HTTP-Client.html#t:Request and https://www.yesodweb.com/book/settings-types